### DISCUSSION TITLE
Docker Installation getting Bad Gateway

### DISCUSSION THREAD

**Discussion by flavioapfernandes** [lowcoder-org/lowcoder#912] [Q&A]
I did a brand new install of the all-in-one image and I'm getting the Bad Gateway error.

![image](https://github.com/lowcoder-org/lowcoder/assets/34424787/bdaf3ffa-c1ee-4325-8f1b-466282e4afc8)

Below is my docker configuration file.

`version: "3"
services:

  # Start Lowcoder (all-in-one)
  lowcoder-api-service:
    image: lowcoderorg/lowcoder-ce:2.4.0
    container_name: lowcoder
    ports:
      - "3000:3000"
      - "3443:3443"
      - "27017:27017"
      - "8080:8080"
    environment:
      # Public base url
      LOWCODER_PUBLIC_URL: "http://192.168.1.226"
      # enable services
      LOWCODER_REDIS_ENABLED: "true"
      LOWCODER_MONGODB_ENABLED: "true"
      # 
      # Set LOWCODER_MONGODB_EXPOSED to "true" and uncomment mongodb port
      # to make internal mongo database accessible from host
      #
      LOWCODER_MONGODB_EXPOSED: "true"
      LOWCODER_API_SERVICE_ENABLED: "true"
      LOWCODER_NODE_SERVICE_ENABLED: "true"
      LOWCODER_FRONTEND_ENABLED: "true"
      # generic parameters
      LOWCODER_PUID: "1000"
      LOWCODER_PGID: "1000"
      # api-service parameters
      LOWCODER_MAX_ORGS_PER_USER: 100
      LOWCODER_MAX_MEMBERS_PER_ORG: 1000
      LOWCODER_MAX_GROUPS_PER_ORG: 100
      LOWCODER_MAX_APPS_PER_ORG: 1000
      LOWCODER_MAX_DEVELOPERS: 50
      #LOWCODER_MONGODB_URL: "mongodb://lowcoder:secret123@mongodb/lowcoder?authSource=admin"
      LOWCODER_MONGODB_URL: "mongodb://192.168.1.226:27017/lowcoder?authSource=admin"
      LOWCODER_REDIS_URL: "redis://192.168.1.226:6379"
      LOWCODER_EMAIL_SIGNUP_ENABLED: "true"
      LOWCODER_EMAIL_AUTH_ENABLED: "true"
      LOWCODER_CREATE_WORKSPACE_ON_SIGNUP: "true"
      #
      # ! PLEASE CHANGE THESE TO SOMETHING UNIQUE !
      # 
      #    LOWCODER_DB_ENCRYPTION_PASSWORD and LOWCODER_DB_ENCRYPTION_SALT is used 
      #    to encrypt sensitive data in database so it is important to change the defaults
      #
      LOWCODER_DB_ENCRYPTION_PASSWORD: "veryuniquePass"
      LOWCODER_DB_ENCRYPTION_SALT: "veryuniquePass"
      LOWCODER_CORS_DOMAINS: "*"
      #
      # API-KEY secret - should be a string of at least 32 random characters
      #    - on linux/mac, generate one eg. with: head /dev/urandom | head -c 30 | shasum -a 256
      #
      LOWCODER_API_KEY_SECRET: "955c0adb354a0520e3acad9c7f072877b2359b000db636b2f82af2d98ac5f456"
      # api and node service parameters
      LOWCODER_API_SERVICE_URL: "http://192.168.1.226:8080"
      LOWCODER_NODE_SERVICE_URL: "http://192.168.1.226:6060"
      # frontend parameters
      LOWCODER_MAX_REQUEST_SIZE: 20m
      LOWCODER_MAX_QUERY_TIMEOUT: 120
      LOWCODER_WORKSPACE_MODE: SAAS
      # Lowcoder notification emails setup
      # LOWCODER_ADMIN_SMTP_HOST: localhost
      # LOWCODER_ADMIN_SMTP_PORT: 587
      #LOWCODER_ADMIN_SMTP_USERNAME: 
      #LOWCODER_ADMIN_SMTP_PASSWORD: 
      #OWCODER_ADMIN_SMTP_AUTH: true
      #OWCODER_ADMIN_SMTP_SSL_ENABLED: false
      #OWCODER_ADMIN_SMTP_STARTTLS_ENABLED: true
      #LOWCODER_ADMIN_SMTP_STARTTLS_REQUIRED: true
      # Email used as sender in lost password email
      #LOWCODER_EMAIL_NOTIFICATIONS_SENDER: info@localhost
    volumes:
      - ./lowcoder-stacks:/lowcoder-stacks
      - ./lowcoder-stacks/assets:/lowcoder/assets
    restart: unless-stopped`

Is there something that I'm not doing correct?

**Comment by FalkWolsky** [comment]
This looks like an all-in-one docker image...?
In case, there is no need to map 8080:8080.
Instead, the API Service will be routed by the port 3000 automatically.

**Comment by flavioapfernandes** [comment]
Changed from all-in-one docker image to multi and is working now.

Thank you for the help.
